src/DiagnosticFilterer.ts

```diff
@@ -100,35 +101,36 @@ export class DiagnosticFilterer {
                 path.isAbsolute(filter.src) ? filter.src : `${this.rootDir}/${filter.src}`
             );
 
-            matchedFilePaths = minimatch.match(Object.keys(this.byFile), src, {
+            const byFileSrcs = Object.keys(this.byFile).map(uri => URI.parse(uri).fsPath);
+            matchedFileUris = minimatch.match(byFileSrcs, src, {
                 nocase: true
-            });
+            }).map(src => util.pathToUri(src).toLowerCase());
 
             //there is no src; this applies to all files
         } else {
-            matchedFilePaths = Object.keys(this.byFile);
+            matchedFileUris = Object.keys(this.byFile);
         }
 
         //filter each matched file
-        for (let filePath of matchedFilePaths) {
-            this.filterFile(filter, filePath);
+        for (let fileUri of matchedFileUris) {
+            this.filterFile(filter, fileUri);
         }
     }
 
-    private filterFile(filter: NormalizedFilter, filePath: string) {
+    private filterFile(filter: NormalizedFilter, fileUri: string) {
         //if the filter is negative, we're turning diagnostics on
         //if the filter is not negative we're turning diagnostics off
         const isSuppressing = !filter.isNegative;
-
+        const lowerFileUri = fileUri.toLowerCase();
```

Comment:
This _probably_ isn't a problem, but it feels strange to call .toLowerCase() on a URI. For example, consider this URI from windows:
`"file:///c%3A/projects/roku/brighterscript1/rootdir/source/common1.brs"`

The `%3A` is some type of hex code representing a colon (for `C:/projects...`). By running .`toLowerCase()`, that `%3a` represents a different value. It feels expensive to parse the URI to get the file path in this situation, so I'm not sure we need to change it, but it just stood out as an interesting problem.

Reply:
I just tested the vscode URI parser, seems it doesn't care about the case of the `a` in `%3A` or `%3a`, so this is most likely fine. 

![image](https://github.com/user-attachments/assets/62943fb3-bcd7-4e47-b4c2-bcfa1e290c6b)